src/System.Memory/tests/Memory/ToString.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.MemoryTests
+{
+    public static partial class MemoryTests
+    {
+        [Fact]
+        public static void ToStringInt()
+        {
+            int[] a = { 91, 92, 93 };
+            var memory = new Memory<int>(a);
+            Assert.Equal("System.Memory<Int32>[3]", memory.ToString());
+            Assert.Equal("System.Memory<Int32>[1]", memory.Slice(1, 1).ToString());
+            Assert.Equal("System.Span<Int32>[3]", memory.Span.ToString());
+        }
+
+        [Fact]
+        public static void ToStringInt_Empty()
+        {
+            var memory = new Memory<int>();
+            Assert.Equal("System.Memory<Int32>[0]", memory.ToString());
+            Assert.Equal("System.Memory<Int32>[0]", Memory<int>.Empty.ToString());
+            Assert.Equal("System.Span<Int32>[0]", memory.Span.ToString());
+        }
+
+        [Fact]
+        public static void ToStringChar()
+        {
+            char[] a = { 'a', 'b', 'c' };
+            var memory = new Memory<char>(a);
+            Assert.Equal("abc", memory.ToString());
+            Assert.Equal("b", memory.Slice(1, 1).ToString());
+            Assert.Equal("abc", memory.Span.ToString());
+        }
+
+        [Fact]
+        public static void ToStringChar_Empty()
+        {
+            var memory = new Memory<char>();
+            Assert.Equal("", memory.ToString());
+            Assert.Equal("", Memory<char>.Empty.ToString());
+            Assert.Equal("", memory.Span.ToString());
+        }
+
+        [Fact]
+        public static void ToStringMemoryFromReadOnlyMemory()
+        {
+            string testString = "abcdefg";
+            Memory<char> memory = MemoryMarshal.AsMemory(testString.AsMemory());
```

Comment:
FWIW an example of where I find the AsMemory naming confusing as a reader of code.